src/main/webapp/app/exercises/programming/participate/programming-submission.service.ts

```diff
@@ -535,22 +535,28 @@ export class ProgrammingSubmissionService implements IProgrammingSubmissionServi
      * @param exerciseId the id of the exercise
      * @param req request parameters
      */
-    getProgrammingSubmissionsForExercise(exerciseId: number, req: { submittedOnly?: boolean; assessedByTutor?: boolean }): Observable<HttpResponse<ProgrammingSubmission[]>> {
+    getProgrammingSubmissionsForExerciseByCorrectionRound(
+        exerciseId: number,
+        req: { submittedOnly?: boolean; assessedByTutor?: boolean },
+        correctionRound?: number,
+    ): Observable<HttpResponse<ProgrammingSubmission[]>> {
+        correctionRound = correctionRound ? correctionRound : 0;
         const options = createRequestOption(req);
         return this.http
-            .get<ProgrammingSubmission[]>(`api/exercises/${exerciseId}/programming-submissions`, {
+            .get<ProgrammingSubmission[]>(`api/exercises/${exerciseId}/round/${correctionRound}/programming-submissions`, {
                 params: options,
                 observe: 'response',
             })
-            .map((res: HttpResponse<ProgrammingSubmission[]>) => this.convertArrayResponse(res));
+            .map((res: HttpResponse<ProgrammingSubmission[]>) => ProgrammingSubmissionService.convertArrayResponse(res));
     }
 
     /**
      * Returns next programming submission without assessment from the server
      * @param exerciseId the id of the exercise
      */
-    getProgrammingSubmissionForExerciseWithoutAssessment(exerciseId: number, lock = false): Observable<ProgrammingSubmission> {
-        let url = `api/exercises/${exerciseId}/programming-submission-without-assessment`;
+    getProgrammingSubmissionForExerciseForCorrectionRoundWithoutAssessment(exerciseId: number, lock = false, correctionRound?: number): Observable<ProgrammingSubmission> {
+        correctionRound = correctionRound ? correctionRound : 0;
```

Comment:
Here as well, `correctionRound = correctionRound ?? 0;`. However it's just a subjective preference since it just looks odd having 3 x correctionRound. If this is juts me just remove the comment